include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_histogram.h

```diff
@@ -415,13 +415,15 @@ struct __histogram_general_private_global_atomics_submitter<__internal::__option
         using _bin_type = oneapi::dpl::__internal::__value_t<_Range2>;
         using _histogram_index_type = ::std::int32_t;
 
-        auto __global_mem_size = __q.get_device().template get_info<sycl::info::device::global_mem_size>();
-        const ::std::size_t __max_segments =
-            ::std::min(__global_mem_size / (__num_bins * sizeof(_bin_type)),
-                       oneapi::dpl::__internal::__dpl_ceiling_div(__n, __work_group_size * __min_iters_per_work_item));
-        const ::std::size_t __iters_per_work_item =
+        const std::uint64_t __global_mem_size = __q.get_device().get_info<sycl::info::device::global_mem_size>();
+        const std::uint64_t __max_groups =
+            oneapi::dpl::__internal::__dpl_ceiling_div(__n, __work_group_size * __min_iters_per_work_item);
+        const std::uint64_t __max_segments =
+            std::min(__global_mem_size / (__num_bins * sizeof(_bin_type)), __max_groups);
+
+        const std::uint64_t __iters_per_work_item =
             oneapi::dpl::__internal::__dpl_ceiling_div(__n, __max_segments * __work_group_size);
-        ::std::size_t __segments =
+        const std::uint64_t __segments =
```

Comment:
optional nitpick: 
The call to `sycl::nd_range` and `sycl::buffer` both accept `sycl::range` as their first argument, so `__segments` will convert this to `std::size_t` anyway.  Perhaps its slightly better to make the conversion explicitly but really I don't think it matters.

Reply:
I've returned std::size_t backed